index.js

```diff
@@ -1,91 +1,158 @@
+  this.hasErrors = function(fileName) {
+    return self.services.getCompilerOptionsDiagnostics()
+            .concat(self.services.getSyntacticDiagnostics(fileName))
+            .concat(self.services.getSemanticDiagnostics(fileName)).length > 0;
+  };
+  
+  this.logErrors = function(fileName, srcDir) {
+      var allDiagnostics = self.services.getCompilerOptionsDiagnostics()
+          .concat(self.services.getSyntacticDiagnostics(fileName))
+          .concat(self.services.getSemanticDiagnostics(fileName));
+          
+      allDiagnostics.forEach(function (diagnostic) {
+          var message = ts.flattenDiagnosticMessageText(diagnostic.messageText, "\n");
+          
+          if (diagnostic.file) {
+              var _a = diagnostic.file.getLineAndCharacterOfPosition(diagnostic.start), line = _a.line, character = _a.character;
+              console.log(chalk.red(diagnostic.file.fileName.replace(srcDir, "") + " (" + (line + 1) + "," + (character + 1) + "): " + message));
+          }
+          else {
+              console.log(chalk.red(message));
+          }
+      });
   }
-  return this._optionsHash;
-};
-
-TypeScript.prototype.cacheKeyProcessString = function(string, relativePath) {
-  return this.optionsHash() + Filter.prototype.cacheKeyProcessString.call(this, string, relativePath);
-};
-
-TypeScript.prototype.processString = function (string, relativePath) {
-  try{
-    return ts.transpileModule(string, {compilerOptions: this.options, fileName: relativePath}).outputText;
-  }catch(e){
-    console.log("TYPESCRIPT ERROR: " + e.message)
-    console.log(e.stack + "\n");
+  
+   this.servicesHost = {
+      getScriptFileNames: function () { return self.rootFilenames; },
+      getScriptVersion: function (fileName) { return self.files[fileName] && self.files[fileName].version.toString(); },
+      getScriptSnapshot: function (fileName) {
+          if (!fs.existsSync(fileName)) {
+              return undefined;
+          }
+          return ts.ScriptSnapshot.fromString(fs.readFileSync(fileName).toString());
+      },
+      getCurrentDirectory: function () { return self.cwd; },
+      getCompilationSettings: function () {
+        return {
+          allowNonTsExtensions: false,
+          diagnostics: true,
+          noImplicitAny: true,
+          //outDir: self.outputPath, TODO: Get this working
+          module: 0,
+          preserveConstEnums: true,
+          removeComments: false,
+          sourceMap: false,
+          target: 2 // ES6
+        };
+      },
+      getDefaultLibFileName: function (options) { return ts.getDefaultLibFilePath(options); },
+  };
+  
+  this.services = ts.createLanguageService(this.servicesHost, ts.createDocumentRegistry());
+  
+  Plugin.call(this, inputNodes, {
+    persistentOutput: true
+  });
+}
 
-    return "";
+TypeScript.prototype.build = function() {
+  var self = this;
+  
+  self.inputPaths.forEach(function(srcDir) {
+    var paths = walkSync(srcDir, { directories: false });
```

Comment:
`walkSync.entries` returns the stat entries, which means you can skip re-stating the files on line 123